mysql-test/suite/innodb/t/autoinc_import.test

```diff
@@ -0,0 +1,133 @@
+ALTER TABLE t10_1b DISCARD TABLESPACE;
+FLUSH TABLES;
+
+# Update the PAGE_ROOT_AUTO_INC field of a few files.
+perl;
+do "$ENV{MTR_SUITE_DIR}/include/crc32.pl";
+sub update_autoinc
+{
+    my ($file, $value) = @_;
+    open(FILE, "+<$file") || die "Unable to open $file";
+    binmode FILE;
+    my $ps= $ENV{PAGE_SIZE};
+    my $page;
+    die "Unable to read $file" unless sysread(FILE, $page, $ps) == $ps;
+    my $full_crc32 = unpack("N",substr($page,54,4)) & 0x10; # FIL_SPACE_FLAGS
+    sysseek(FILE, 3*$ps, 0) || die "Unable to seek $file\n";
+    die "Unable to read $file" unless sysread(FILE, $page, $ps) == $ps;
+    substr($page,56,8)=pack("NN",0,$value);
+    my $polynomial = 0x82f63b78; # CRC-32C
+    if ($full_crc32) {
+        my $ck = mycrc32(substr($page, 0, $ps-4), 0, $polynomial);
+        substr($page, $ps-4, 4) = pack("N", $ck);
+    }
+    else
+    {
+        my $ck= pack("N",mycrc32(substr($page, 4, 22), 0, $polynomial) ^
+                     mycrc32(substr($page, 38, $ps - 38 - 8), 0, $polynomial));
+        substr($page,0,4)=$ck;
+        substr($page,$ps-8,4)=$ck;
+    }
+    sysseek(FILE, 3*$ps, 0) || die "Unable to rewind $file\n";
+    syswrite(FILE, $page, $ps)==$ps || die "Unable to write $file\n";
+    close(FILE) || die "Unable to close $file";
+}
+update_autoinc("$ENV{DATADIR}/test/t.ibd", 3);
+update_autoinc("$ENV{DATADIR}/test/tb.ibd", 346);
+update_autoinc("$ENV{DATADIR}/test/tt.ibd", 128);
+EOF
+
+--remove_file $DATADIR/test/t5_7.frm
+--remove_file $DATADIR/test/t5_7b.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_57.frm $DATADIR/test/t5_7.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_57.frm $DATADIR/test/t5_7b.frm
+--remove_file $DATADIR/test/t10_1.frm
+--remove_file $DATADIR/test/t10_1b.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_101.frm $DATADIR/test/t10_1.frm
+--copy_file $MYSQLTEST_VARDIR/std_data/autoinc_import_101.frm $DATADIR/test/t10_1b.frm
+--copy_file $DATADIR/test/t.ibd $DATADIR/test/t5_7.ibd
+--copy_file $DATADIR/test/tb.ibd $DATADIR/test/t5_7b.ibd
+--copy_file $DATADIR/test/t.ibd $DATADIR/test/t10_1.ibd
+--copy_file $DATADIR/test/tb.ibd $DATADIR/test/t10_1b.ibd
+--move_file $DATADIR/test/t.ibd $DATADIR/test/t1.ibd
+--move_file $DATADIR/test/tb.ibd $DATADIR/test/t1b.ibd
+--move_file $DATADIR/test/tt.ibd $DATADIR/test/t1t.ibd
+
+ALTER TABLE t1 IMPORT TABLESPACE;
+ALTER TABLE t1b IMPORT TABLESPACE;
+ALTER TABLE t1t IMPORT TABLESPACE;
+ALTER TABLE t5_7 IMPORT TABLESPACE;
+ALTER TABLE t5_7b IMPORT TABLESPACE;
+ALTER TABLE t10_1 IMPORT TABLESPACE;
+ALTER TABLE t10_1b IMPORT TABLESPACE;
+
+# Restart, so that the InnoDB tables will be loaded into the data dictionary.
+--let $restart_parameters=--innodb-read-only --read-only
+--source include/restart_mysqld.inc
+
+# FIXME: If we executed CHECK TABLE here, it would upgrade the mysql_version
+# in the .frm files despite --read-only!
+# CHECK TABLE t1, t1b, t1t, t5_7, t5_7b, t10_1, t10_1b;
```

Comment:
It would be better to mention [MDEV-33308](https://jira.mariadb.org/browse/MDEV-33308) (which I just filed) in the comment.

Reply:
Better to apply my check table patch to be part of this one so that you can uncomment the above check tables.

Reply:
This is part of the revised commit efb8c05d352fbf27849866ad8ba8d2c5f822c03f.